### PR TITLE
Remove non generic image

### PR DESCRIPTION
**Pull request by JimBobSquarePants**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/JimBobSquarePants/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practise as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
This PR does two things fixing #208 

1. Removes the non-generic `Image`, `Brushes`, `Brush`, `Pens`, and `Pen `classes in favour of their generic `TPixel` equivalents.
2. Promotes `Rgba32` to the root `ImageSharp` namespace to improve visibility.

Most developers, since `var` is so common, will now do this to load/manipulate images.
```
using (var image = Image<Rgba32>.Load("foo.jpg"))
{
    image.Resize(image.Width / 2, image.Height / 2)
         .Grayscale()
         .Save("bar.jpg"); 
}
```

### CODE REVIEW

**Comment by tocsoft** [comment]
In my opinion feels nicer to type `Image.Load<TPixel>(...)` than `Image<TPixel>.Load(...)`

**Comment by JimBobSquarePants** [comment]
Yeah, you're right. That would better reflect existing types like `Enumerable` also.

**Comment by tocsoft** [comment]
Maybe even just leave `Image<TPixel>.Load()` in but also add `Image.Load<TPixel>()` versions in that just call down into the first.

**Comment by JimBobSquarePants** [comment]
Thanks! @dlemstra

**Comment by dlemstra** [comment]
I have to agree with @tocsoft and I also like `Image.Load<TPixel>()` better. Maybe we should just have the following options available:

- `Image<TPixel>.Load()`
- `Image.Load<TPixel>()`
- `Image.Load()` = `Image.Load<Rgba32>()`

I would also like the last one to be available because most people will probably only use the Rgba32 variant.

**Comment by JimBobSquarePants** [comment]
Almost finished making the other changes but the last one would require me to create an overload for every overload of Load. Even saying sounds crazy. Devs should use the generic method.

**Comment by dlemstra** [comment]
I do think we only need that for two overloads. The one with only a file name and the one with only a stream. And if you want more you should use one of the generic overloads.

**Comment by JimBobSquarePants** [comment]
@dlemstra ok coke. Only cos you're awesome 😄

**Comment by dlemstra** [comment]
That should be extra super awesome instead of just awesome. And thanks great leader!

**Comment by JimBobSquarePants** [comment]
OK. So adding the `Rgba32` overloads for all the `Load` methods was less work than I thought.

Rgba32 can use either `Image.Load()`or `Image.Load<Rgba32>()`

All other pixel formats would use `Image.Load<TPixel>()`

I think we're good to go now.

File: src/ImageSharp/Image/Image.Create.cs [hunk 1]
```diff
@@ -7,56 +7,46 @@ namespace ImageSharp
...
         /// <param name="configuration">
         /// The configuration providing initialization code which allows extending the library.
         /// </param>
+        /// <typeparam name="TPixel">The pixel format.</typeparam>
         /// <returns>
-        /// A new <see cref="Image{TPixel}"/> unless <typeparamref name="TPixel"/> is <see cref="Rgba32"/> in which case it returns <see cref="Image" />
+        /// A new <see cref="Image{TPixel}"/>.
         /// </returns>
-        internal static Image<TPixel> Create<TPixel>(int width, int height, ImageMetaData metadata, Configuration configuration)
+        internal static Image<TPixel> Create<TPixel>(int width, int height, Configuration configuration)
```
Comment (tocsoft) [inline src/ImageSharp/Image/Image.Create.cs]: we can get rid of these now, now we don't have to do the `if (typeof(TPixel) == typeof(Rgba32))` thing, and just call the constructor directly from the `IImageDecoder`s

**Comment by JimBobSquarePants** [comment]
Thanks for finishing this up guys, was way past my bedtime!

**Pull request merged**